src/app/shared/palette/complex/read-complex-field.component.ts

```diff
@@ -1,9 +1,30 @@
 import { Component } from '@angular/core';
 import { AbstractFieldReadComponent } from '../base-field/abstract-field-read.component';
+import { CaseField } from '../../domain/definition/case-field.model';
 
 @Component({
   selector: 'ccd-read-complex-type-field',
   templateUrl: './read-complex-field.html',
   styleUrls: ['./read-complex-field.scss']
 })
-export class ReadComplexFieldComponent extends AbstractFieldReadComponent {}
+export class ReadComplexFieldComponent extends AbstractFieldReadComponent {
+
+  getComplexFields() {
```

Comment:
not sure if this is an issue but if we make a copy of the complex fields, and the from the ui we update the copied fields, does the parent type sees the update?